configure.ac

```diff
@@ -102,13 +102,15 @@ AM_INIT_AUTOMAKE([foreign dist-zip subdir-objects])
 AC_CONFIG_SRCDIR([include/netcdf.h])
 
 # Figure out platforms of special interest
-case "`uname`" in
-  CYGWIN*) ISCYGWIN=yes;;
-  Darwin*) ISOSX=yes;;
-  WIN*) ISMSVC=yes;;
-  MINGW*) ISMINGW=yes;;
-  MSYS*) ISMINGW=yes;;
-esac
+AC_CANONICAL_HOST
```

Comment:
I vaguely recall that we had problems in the past using AC_CANONICAL_HOST.
Wish I could remember the details.

Reply:
It won't auto-discover MSVC, but it should help with #2396, since the current assumption made in using `uname` directly is that nobody will want to cross-compile at any time for any reason